Src/ILGPU/IR/Construction/Method.Builder.cs

```diff
@@ -608,17 +623,21 @@ private void UpdateParameters()
             protected override void Dispose(bool disposing)
             {
                 // Update parameter bindings
-                UpdateParameters();
+                if (builderCompleted)
```

Comment:
QUESTION: Should we add an internal `Debug.Assert` to this spot which checks that the builder has been completed? This helps us to discover programming errors at development time while still preserving the ability to throw "nice" exceptions.

Reply:
Done 👍